[CPP] : Missing variable declarations

Ignore the nonsense functionality of the shader, it's just a simple way to reproduce the problem:
```
#version 450 core
layout(std430, binding = 0) readonly buffer Indices {
    uint indices[];
};
layout(std430, binding = 1) readonly buffer Values {
    float values[];
};
layout(std430, binding = 2) buffer Output {
    float output_value;
};

layout(binding=3) uniform UniformData {
    float value;
};

const ivec2 edges[3] = ivec2[3]( ivec2(0,1), ivec2(1,2), ivec2(2,0) );
void main()
{
    uint idx = gl_GlobalInvocationID.x;
    for( int i=0; i<3; ++i )
    {
        if( (value - values[indices[idx+edges[i].s]]) * (values[indices[idx+edges[i].t]] - value) >= 0 )
        {
            output_value = value;
        }
    }
}
```

Compiling that to SPIR-V is just fine and valid, the issue is then running through spirv-cross to generate a cpp. The `edges` variable is not declared so when compiling the resultant cpp you see the expected error:
```
test.comp.spv.cpp: In member function ‘void Impl::Shader::main()’:
test.comp.spv.cpp:66:73: error: ‘_53’ was not declared in this scope
                 if (((_32.value - _39.values[_43.indices[idx + uint32_t(_53[i].x)]]) * (_39.values[_43.indices[idx + uint32_t(_53[i].y)]] - _32.value)) >= 0.0f)
```

`_53` is the edges variable and looking in the cpp the declaration is missing, declaring it as a local variable in the main function yields the same result. Adding it manually to the cpp as a std::array allows it to build and run just fine.

This data is constant and small so the easy workaround is supplying it as a storage/uniform/sampler or push constant but it does expose an issue.

--- a/spirv_cpp.cpp
+++ b/spirv_cpp.cpp
@@ -143,6 +143,30 @@ void CompilerCPP::emit_block_struct(SPIRType &type)
 
 void CompilerCPP::emit_resources()
 {
+	for (auto &id : ir.ids)
+	{
+		if (id.get_type() == TypeConstant)
+		{
+			auto &c = id.get<SPIRConstant>();
+
+			bool needs_declaration = c.specialization || c.is_used_as_lut;
+
+			if (needs_declaration)
+			{
+				if (!options.vulkan_semantics && c.specialization)
+				{
+					c.specialization_constant_macro_name =
+					    constant_value_macro_name(get_decoration(c.self, DecorationSpecId));
+				}
+				emit_constant(c);
+			}
+		}
+		else if (id.get_type() == TypeConstantOp)
+		{
+			emit_specialization_constant_op(id.get<SPIRConstantOp>());
+		}
+	}
+
 	// Output all basic struct types which are not Block or BufferBlock as these are declared inplace
 	// when such variables are instantiated.
 	for (auto &id : ir.ids)

